client/models.py

```diff
@@ -281,7 +328,34 @@ def image_download_url(self):
             return reverse('download_certificate', kwargs={'pk': self.id})
 
 
+class HelpRequestVolunteerManager(models.Manager):
+    def _coord_distance(self, p1, p2):
+        ''' Haversine Formula '''
+        earth_radius =  6371
+        lat1 = math.radians(p1[0])
+        lat2 = math.radians(p2[0])
+        dLat = math.radians(p2[0]-p1[0])
+        dLon = math.radians(p2[1]-p1[1])
+
+        a = math.sin(dLat/2) ** 2 + math.cos(lat1) * math.cos(lat2) * math.sin(dLon/2)**2;
+        c = 2 * math.atan2(math.sqrt(a), math.sqrt(1-a))
+        return earth_radius * c
+
+    def all_by_distance(self, h_coord):
+        volunteers = []
+        for v in Volunteer.objects.all():
```

Comment:
You iterate through all volunteers here - is it possible to move the calculation to be on DB side instead of python? This could be a real RAM and CPU choker if we keep it this way.
We need to find a way to only get the relevant portion of volunteers from the DB.